dotnet/samples/KernelSyntaxExamples/Example57_FunctionEventHandlers.cs

```diff
@@ -91,6 +93,59 @@ void MyPostExecutionHandler(object? sender, FunctionInvokedEventArgs e)
         Console.WriteLine($"Function Result: {result.GetValue<string>()}");
     }
 
+    private static async Task GetRenderedPromptAsync()
+    {
+        Console.WriteLine("\n======== Get Rendered Prompt ========\n");
+
+        IKernel kernel = new KernelBuilder()
+            .WithLoggerFactory(ConsoleLogger.LoggerFactory)
+            .WithOpenAIChatCompletionService(
+                modelId: s_openAIModelId!,
+                apiKey: s_openAIApiKey!)
+            .Build();
+
+        const string FunctionPrompt = "Write a random paragraph about: {{$input}}.";
+
+        var excuseFunction = kernel.CreateSemanticFunction(
+            FunctionPrompt,
+            pluginName: "MyPlugin",
+            functionName: "Excuse",
+            requestSettings: new OpenAIRequestSettings() { MaxTokens = 100, Temperature = 0.4, TopP = 1 });
+
+        void MyPreHandler(object? sender, FunctionInvokingEventArgs e)
+        {
+            Console.WriteLine($"{e.FunctionView.PluginName}.{e.FunctionView.Name} : Pre Execution Handler - Triggered");
+
+            if (e.TryGetRenderedPrompt(out var prompt))
```

Comment:
We could potentilly pass any data to Metadata dictionary, I'm not passing anything extra as this would be Out of Scope of this PR. We can add those once the LLM Multi Model/Model support is in place with the new IAIService interfaces.